Initial support for generating go protos (#214)
abcxyz/abc: t/proto_template/README.md

# Context

> You must have various tools installed in order to generate protos. The tool does not distinguish between language types and instead seeks to generate protos for a variety of supported languages. Currently only go is supported. Node will be supported in the near future.

Reviewer: ```suggestion
Every language has its own tools for generating protos. The tool will generate protos for a variety of supported languages. Currently only go is supported. Node will be supported in the near future.
```

## Tools

Install `protoc`, a compiler for generating language specific protos based on .proto files. 

```
  brew install protobuf # for MacOS users

>   apt install -y protobuf-compiler # for Linux users

Reviewer: Nit: apt is only for debian-based distros, other linux distros have other package managers (dnf/yum/pacman)

```

### Go

Ensure you have go installed, if not run the following

```

>   brew install go

Reviewer: Should also have linux instructions

Reviewer 2: https://docs.brew.sh/Homebrew-on-Linux 🙃
